doc/docs.md

```diff
@@ -2499,6 +2499,60 @@ fn main() {
 }
 ```
 
+### Function Types
+
+Function signatures can be declared as types - for example:
+
+```v
+type Filter = fn (string) string
+```
+
+This works like any other type - for example, a function can accept an argument of a function type:
+
+```
+fn filter(s string, f Filter) string {
+  return f(s)
+}
+```
+
+V has duck-typing, so functions don't need to declare compatibility with a function type - they just have to be compatible:
+
+```
+fn uppercase(s string) string {
+	return s.to_upper()
+}
+```
+
+Compatible functions can be explicitly cast to a function type:
+
+```v
+my_filter := Filter(uppercase)
+```
```

Comment:
Such code examples should be marked with ```v oksyntax , otherwise `v check-md doc/docs.md` will complain, because it will try to extract them as separate .v files, and compile them, to verify that they do work, and that compilation will fail (the code is incomplete).
See also `./v help check-md` .